content/blog/2019/08/2019-08-17-folder-auth-plugin.adoc

```diff
@@ -0,0 +1,111 @@
+
+image::/images/post-images/gsoc-folder-auth-plugin/folder-auth.png[Screenshot of the Folder Auth Plugin]
+
+This plugin was inspired by the plugin:role-strategy[Role Strategy Plugin]
+and brings about performance improvements and makes managing roles much easier.
+The plugin supports three types of roles which are applicable at different places
+in Jenkins.
+
+* Global Roles: applicable everywhere in Jenkins
+* Agent Roles: restrict permissions for multiple agents connected to your instance
+* Folder Roles: applicable to multiple jobs organized inside folders
+
+This plugin, unlike the Role Strategy plugin, does not use regular expressions
+for finding matching projects and agents giving us performance improvements
+and makes administrators' lives easier. To reduce the number of roles required
+to be managed, permissions given to a folder through a folder role get inherited
+to all of its children. This is useful for giving access to multiple projects
+through a single role. Similarly, an agent role can be applied to multiple agents
+and assigned to multiple users.
+
+==== Performance Improvements over Role Strategy Plugin
+
+This plugin is designed to outperform Role Strategy Plugin in permission
+checks. The improvements were measured using the 
+link:blog/2019/06/21/performance-testing-jenkins/[micro-benchmark framework]
+I had created during the first phase of my GSoC project.
+Benchmarks for identical configurations for both plugin show that the
+permissions check are up to 934x faster for 500 global roles when compared to
+the global roles from the Role Strategy 2.13, which in itself contains several
+performance improvements. Comparing folder roles with Role Strategy's project
+roles, a permission check for access to a job almost 15x faster.
+
+==== Jenkins Configuration as Code Support
+
+The plugin supports Jenkins Configuration-as-Code so you can configure permissions
+without going through the Web UI. A YAML configuration looks like this:
+
+[source, yaml]
+----
+jenkins:
+  authorizationStrategy:
+    folderBased:
+      agentRoles:
+        - agents:
+            - "agent1"
+          name: "agentRole1"
+          permissions:
+            - id: "hudson.model.Computer.Configure"
+            - id: "hudson.model.Computer.Disconnect"
+          sids:
+            - "user1"
+      folderRoles:
+        - folders:
+            - "root"
+          name: "viewRoot"
+          permissions:
+            - id: "hudson.model.Item.Read"
+          sids:
+            - "user1"
+      globalRoles:
+        - name: "admin"
+          permissions:
+            - id: "hudson.model.Hudson.Administer"
+              # ...
+          sids:
+            - "admin"
+        - name: "read"
+          permissions:
+            - id: "hudson.model.Hudson.Read"
+          sids:
+            - "user1" 
+----
+
+==== REST APIs with Swagger support
+
+The plugin provides REST APIs for managing roles with OpenAPI specifications
+through Swagger.json.
```

Comment:
Add some extra text in addition to the documentation screenshot? The SwaggerHub reference could be also in the first paragraph before the screenshot